commit 40d3f295b5feda409784e569550057b5fbc2a295
Author: Vladimir Oltean <vladimir.oltean@nxp.com>
Date:   Sun Feb 14 00:37:56 2021 +0200

    net: mscc: ocelot: use common tag parsing code with DSA
    
    The Injection Frame Header and Extraction Frame Header that the switch
    prepends to frames over the NPI port is also prepended to frames
    delivered over the CPU port module's queues.
    
    Let's unify the handling of the frame headers by making the ocelot
    driver call some helpers exported by the DSA tagger. Among other things,
    this allows us to get rid of the strange cpu_to_be32 when transmitting
    the Injection Frame Header on ocelot, since the packing API uses
    network byte order natively (when "quirks" is 0).
    
    The comments above ocelot_gen_ifh talk about setting pop_cnt to 3, and
    the cpu extraction queue mask to something, but the code doesn't do it,
    so we don't do it either.
    
    Signed-off-by: Vladimir Oltean <vladimir.oltean@nxp.com>
    Reviewed-by: Florian Fainelli <f.fainelli@gmail.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

diff --git a/drivers/net/ethernet/mscc/ocelot.h b/drivers/net/ethernet/mscc/ocelot.h
index c485795c606b..db6b1a4c3926 100644
--- a/drivers/net/ethernet/mscc/ocelot.h
+++ b/drivers/net/ethernet/mscc/ocelot.h
@@ -32,15 +32,6 @@
 
 #define OCELOT_PTP_QUEUE_SZ	128
 
-struct frame_info {
-	u32 len;
-	u16 port;
-	u16 vid;
-	u8 tag_type;
-	u16 rew_op;
-	u32 timestamp;	/* rew_val */
-};
-
 struct ocelot_port_tc {
 	bool block_shared;
 	unsigned long offload_cnt;